test_elasticsearch_serverless/test_server/test_rest_api_spec.py

```diff
@@ -568,20 +568,25 @@ def remove_implicit_resolver(cls, tag_to_remove):
         # Each file may have a "test" named 'setup' or 'teardown',
         # these sets of steps should be run at the beginning and end
         # of every other test within the file so we do one pass to capture those.
-        setup_steps = teardown_steps = None
+        requires = setup_steps = teardown_steps = None
         test_numbers_and_steps = []
         test_number = 0
 
         for yaml_test in yaml_tests:
             test_name, test_step = yaml_test.popitem()
-            if test_name == "setup":
+            if test_name == "requires":
+                requires = test_step
+            elif test_name == "setup":
                 setup_steps = test_step
             elif test_name == "teardown":
                 teardown_steps = test_step
             else:
                 test_numbers_and_steps.append((test_number, test_step))
                 test_number += 1
 
+        if not requires["serverless"]:
```

Comment:
Can a test not have a `requires` section, or have it, but not have a `serverless` key in it? Not sure if that is possible, but I think it would make sense to protect against a crash here.

Reply:
It's not possible. If it happens, I'd rather crash loudly than making a wrong choice (offensive programming vs. defensive programming). After all, we fully control the tests here.